pkg/api/auth.go

```diff
@@ -253,6 +254,32 @@ func (a *API) authMiddleware(allowedUsers []string) gin.HandlerFunc {
 		}
 
 		c.Set("user", user)
+		c.Set("token", session.AccessToken)
+		c.Next()
+	}
+}
+
+func (a *API) adminAuthMiddleware() gin.HandlerFunc {
+	return func(c *gin.Context) {
+		value, _ := c.Get("user")
+		user, pass := value.(*auth.User)
+		if !pass {
+			a.log.Error("User info missing")
+			c.Redirect(http.StatusSeeOther, "/")
+			return
+		}
+		value, _ = c.Get("token")
+		token, pass := value.(string)
```

Comment:
Should either ignore pass with _  or check and log error and return if the type assertion fails